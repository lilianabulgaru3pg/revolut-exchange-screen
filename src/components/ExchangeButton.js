import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Fab from '@material-ui/core/Fab';
import PropTypes from 'prop-types';

// todo addtheme
const useStyles = makeStyles({
  grid: {
    marginTop: '5%',
  },
  fab: {
    margin: '10%',
    paddingLeft: '15%',
    paddingRight: '15%',
  },
});

// todo create generic btn
const ExchangeButton = ({ onExchange, disabled = false, title }) => {
  const classes = useStyles();
  return (
    <Grid container justify="center" className={classes.grid}>
      <Fab
        variant="extended"
        color="secondary"
        aria-label="exchange"
        className={classes.fab}
        onClick={() => onExchange()}
        disabled={disabled}
      >
        {title}
      </Fab>
    </Grid>
  );
};

ExchangeButton.propTypes = {
  onExchange: PropTypes.func.isRequired,
  disabled: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired,
};

export default ExchangeButton;
