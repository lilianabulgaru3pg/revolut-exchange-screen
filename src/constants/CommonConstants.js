const CURRENCIES = {
  EUR: 'EUR',
  USD: 'USD',
  GBP: 'GBP',
};

const SYMBOL = {
  EUR: '€',
  USD: '$',
  GBP: '£',
};

const EB_BASE_URL = 'https://api.exchangeratesapi.io';

const ENABLE_EXCHANGE_BTN = 'ENABLE_EXCHANGE_BTN';
const DISABLE_EXCHANGE_BTN = 'DISABLE_EXCHANGE_BTN';

const DEFAULT_TO_CURRENCY = 'EUR';
const USER_DEFAULT_BASE_CURRENCY = 'GBP';

const SELECT_BASE_CURRENCY = 'SELECT_BASE_CURRENCY';
const SELECT_TO_CURRENCY = 'SELECT_TO_CURRENCY';

const availableBalace = 500;
const exchangeTitle = 'Exchange';

const inputFromType = 'from';
const inputToType = 'to';

const FETCH_INTERVAL = 10000;

export {
  SYMBOL,
  CURRENCIES,
  FETCH_INTERVAL,
  DEFAULT_TO_CURRENCY,
  USER_DEFAULT_BASE_CURRENCY,
  SELECT_BASE_CURRENCY,
  SELECT_TO_CURRENCY,
  availableBalace,
  exchangeTitle,
  ENABLE_EXCHANGE_BTN,
  DISABLE_EXCHANGE_BTN,
  EB_BASE_URL,
  inputFromType,
  inputToType,
};
