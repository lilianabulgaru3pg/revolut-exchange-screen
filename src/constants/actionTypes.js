export const SELECT_BASE_CURRENCY = 'SELECT_BASE_CURRENCY';
export const SELECT_TO_CURRENCY = 'SELECT_TO_CURRENCY';

export const SELECT_BASE_SYMBOL = 'SELECT_BASE_SYMBOL';
export const SELECT_TO_SYMBOL = 'SELECT_TO_SYMBOL';

export const DISABLE_EXCHANGE_BTN = 'DISABLE_EXCHANGE_BTN';
export const ENABLE_EXCHANGE_BTN = 'ENABLE_EXCHANGE_BTN';

export const EXCHANGE = 'EXCHANGE';
export const EXCHANGE_BALANCE = 'EXCHANGE_BALANCE';

export const RATES_FETCH = 'RATES_FETCH';
export const RATES_FETCH_ERROR = 'RATES_FETCH_ERROR';

export const RATES_ADD = 'RATES_ADD';

export const INPUT_FROM = 'INPUT_FROM';
export const INPUT_TO = 'INPUT_TO';
export const DISPLAY_INPUT_FROM = 'CHANGE_INPUT_FROM';
export const DISPLAY_INPUT_TO = 'CHANGE_INPUT_TO';
export const RESET_INPUTS = 'RESET_INPUTS';
