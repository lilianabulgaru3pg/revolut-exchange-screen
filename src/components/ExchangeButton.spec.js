import React from 'react';
import renderer from 'react-test-renderer';
import ExchangeButton from './ExchangeButton';
import { exchangeTitle } from '../constants/CommonConstants';

describe('ExchangeButton Component', () => {
  test('snapshot renders', () => {
    const component = renderer.create(
      <ExchangeButton
        onExchange={jest.fn()}
        disabled={false}
        title={exchangeTitle}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
