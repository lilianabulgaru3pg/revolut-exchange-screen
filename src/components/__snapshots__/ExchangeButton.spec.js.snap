// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeButton Component snapshot renders 1`] = `
<div
  className="MuiGrid-root makeStyles-grid-1 MuiGrid-container MuiGrid-justify-xs-center"
>
  <button
    aria-label="exchange"
    className="MuiButtonBase-root MuiFab-root makeStyles-fab-2 MuiFab-extended MuiFab-secondary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiFab-label"
    >
      Exchange
    </span>
  </button>
</div>
`;
